app/api/competitions/route.js

```diff
@@ -0,0 +1,27 @@
+import dbConnect from '@/utils/dbConnects';
+import Competition from '../models/Competition';
+import { verifyToken } from '@/utils/auth';
+
+export async function GET() {
+  await dbConnect();
+  const competitions = await Competition.find().sort({ createdAt: -1 });
+  return Response.json(competitions);
+}
+
+export async function POST(req) {
+  try {
+    await dbConnect();
+    verifyToken(req);
```

Comment:
We should have a middleware to verify the token. This can be done in a followup PR